composer/core/data_spec.py

```diff
@@ -244,7 +244,16 @@ def _default_get_num_samples_in_batch(self, batch: Batch) -> int:
                                          '`get_num_samples_in_batch(your_batch) -> int` method.')
                     dim0_sizes.append(t.shape[0])
         elif isinstance(batch, dict):
-            dim0_sizes = [t.shape[0] for t in batch.values()]
+            for t in batch.values():
+                if isinstance(t, torch.Tensor):
+                    dim0_sizes.append(t.shape[0])
+                elif isinstance(t, list):
+                    dim0_sizes.append(len(t))
+                else:
+                    raise ValueError('Unable to determine the batch size, batch contains'
+                                     f'an element of type {type(t)}, which does not have a'
```

Comment:
since the code above checks types, not whether `t` has a shape, can you update the error message to say what it is actually checking (is tensor or list)